java/ql/test/kotlin/library-tests/parameter-defaults/test.kt

```diff
@@ -161,3 +161,15 @@ class VisibilityTests {
   private fun i(x: Int, y: Int = 0) = x + y
 
 }
+
+class TestGenericUsedWithinDefaultValue<T> {
+
+  // This tests parameter erasure works properly: we should notice that here the type variable T
+  // isn't used in the specialisation TestGenericUsedWithinDefaultValue<String>, but it can be
+  // cited in contexts like "the signature of the source declaration of 'TestGenericUsedWithinDefaultValue<String>.f(String)' is 'f(T)'",
+  // not 'f(Object)' as we might mistakenly conclude if we're inappropriately erasing 'T'.
```

Comment:
How would I see this in QL? Would this mean that `Callable.getSourceDeclaration().getSignature()` would return `ident(T)` on the target of `.ident("Hello world")`?

Reply:
Yes. Without the extractor change we'd produce an inconsistency, trying to target a source-decl ident(Object) that doesn't exist -- you should be able to see it by reverting the .kt change and running the test.